compiler/src/main/kotlin/dev/zacsweers/metro/compiler/fir/FirAccessorOverrideStatusTransformer.kt

```diff
@@ -119,4 +122,18 @@ internal class FirAccessorOverrideStatusTransformer(session: FirSession) :
     if (!needsOverride) return status
     return status.copy(isOverride = true)
   }
+
+  private data class Accessor(
+    val name: Name,
+    val isFunction: Boolean,
+    val returnType: ConeKotlinType
+  )
```

Comment:
Introduced a local class to represent the accessor information, which is used for storing in a set and comparison.